pkg/jobrunaggregator/releasebigqueryloader/release_uploader.go

```diff
@@ -0,0 +1,225 @@
+	gcsClient  jobrunaggregatorlib.CIGCSClient
+	httpClient *http.Client
+
+	releaseInserter *bigquery.Inserter
+	releaseTable    *bigquery.Table
+
+	releaseJobRunInserter *bigquery.Inserter
+	releaseJobRunTable    *bigquery.Table
+
+	releases  []string
+	ciDataSet *bigquery.Dataset
+}
+
+func (r *allReleaseUploaderOptions) Run(ctx context.Context) error {
+	if err := r.findOrCreateTable(ctx, jobrunaggregatorlib.ReleaseTableName); err != nil {
+		return errors.Wrapf(err, "could not find or create %s table", jobrunaggregatorlib.ReleaseTableName)
+	}
+
+	if err := r.findOrCreateTable(ctx, jobrunaggregatorlib.ReleaseJobRunTableName); err != nil {
+		return errors.Wrapf(err, "could not find or create %s table", jobrunaggregatorlib.ReleaseJobRunTableName)
+	}
+
+	releaseTagSet, err := r.ciDataClient.ListReleaseTags(ctx)
+	if err != nil {
+		return err
+	}
+
+	for _, release := range r.releases {
+		fmt.Fprintf(os.Stderr, "Fetching release %s from release controller...\n", release)
+		tags := r.fetchReleaseTags(release)
+
+		for _, tag := range tags.Tags {
+			fmt.Fprintf(os.Stderr, "Fetching tag %s from release controller...\n", tag.Name)
+			releaseDetails := r.fetchReleaseDetails(release, tag)
+			bigQueryReleaseDetails := releaseDetailsToBigQuery(tag, releaseDetails)
+
+			// We skip releases that aren't fully baked, or already in the big query tables:
+			if _, ok := releaseTagSet[bigQueryReleaseDetails.ReleaseTag]; ok {
+				continue
+			}
+
+			if bigQueryReleaseDetails.Phase == "Ready" || !bigQueryReleaseDetails.ChangeLog.Valid {
+				continue
+			}
+
+			if err := r.releaseInserter.Put(ctx, bigQueryReleaseDetails); err != nil {
+				return errors.Wrapf(err, "could not insert release details for %s", tag.Name)
+			}
+			for _, run := range releaseJobRunsToBigQuery(releaseDetails) {
+				if err := r.releaseJobRunInserter.Put(ctx, run); err != nil {
+					return errors.Wrapf(err, "could not insert job run %s", run.Name)
+				}
+			}
+		}
+
+	}
+
+	return nil
+}
+
+func (r *allReleaseUploaderOptions) fetchReleaseDetails(release string, tag ReleaseTag) ReleaseDetails {
+	releaseDetails := ReleaseDetails{}
+	url := fmt.Sprintf("https://amd64.ocp.releases.ci.openshift.org/api/v1/releasestream/%s/release/%s", release, tag.Name)
+
+	resp, err := r.httpClient.Get(url)
+	if err != nil {
+		panic(err)
+	}
+	defer resp.Body.Close()
+	if err := json.NewDecoder(resp.Body).Decode(&releaseDetails); err != nil {
+		panic(err)
+	}
+	return releaseDetails
+}
+
+func (r *allReleaseUploaderOptions) fetchReleaseTags(release string) ReleaseTags {
+	tags := ReleaseTags{}
+
+	resp, err := r.httpClient.Get(fmt.Sprintf("https://amd64.ocp.releases.ci.openshift.org/api/v1/releasestream/%s/tags", release))
+	if err != nil {
+		panic(err)
+	}
+	defer resp.Body.Close()
+	if err := json.NewDecoder(resp.Body).Decode(&tags); err != nil {
+		panic(err)
+	}
+	return tags
+}
+
+func (r *allReleaseUploaderOptions) findOrCreateTable(ctx context.Context, tableName string) error {
```

Comment:
I prefer leaving the definition of the table distinct from its usage.

Reply:
Do you want a command to explicitly create tables? Or maintain the behavior of creating them but split the logic to two functions?